upgrade_tests/test_scenarios.py

```diff
@@ -286,3 +286,246 @@ def test_post_user_scenario_bug_1429201(self):
         self.assertIsNotNone(container_ids)
         self.assertIn('Error', result_fail.values()[0])
         self.assertIn('Complete', result_pass.values()[0])
+
+
+class Scenario_capsule_sync(TestCase):
+    """The test class contains pre-upgrade and post-upgrade scenarios to test if
+    package added to satellite preupgrade is synced to capsule post upgrade.
+
+    Test Steps:
+
+    1. Before Satellite upgrade, Sync a repo/rpm in satellite.
+    2. Upgrade satellite/capsule.
+    3. Run capsule sync post upgrade.
+    4. Check if the repo/rpm is been synced to capsule.
+
+    """
+    sat_host = get_satellite_host()
+    env.host_string = sat_host
+    env.user = 'root'
+    hammer.set_hammer_config()
```

Comment:
Why I dont because I need that things even before setup part, during class initialization itself. If I put that in setup then my test doesnt works. 
Also I cant put everything in setup because I cant use them outside of setup def.